Server/src/models/DatasetModels/DatasetCommonModel.ts

```diff
@@ -43,4 +43,33 @@ export class DatasetCommonModel {
         let verification = await this.connection.manager.findOne(Dataset, { id: id });
         return verification
     }
+
+    async verifyUnapprovedDatasetUploader(id: number, userId: number): Promise<any> {
+        let upload = await this.connection.createQueryBuilder(Unapproveddatasets, 'unapproved_datasets')
+            .select('dataset.uploaderId', 'uploaderId')
+            .addSelect('unapproved_datasets.isFlagged', 'isFlagged')
+            .innerJoin(Dataset, 'dataset', 'dataset.id = unapproved_datasets.datasetId')
+            .where('dataset.id = :id', { id: id })
+            .getRawOne();
+        if (!upload) {
+            return "No such unapproved data set exists!"
+        }
+        else if (upload.uploaderId != userId) {
+            return "User ID does not match uploader ID!"
+        }
+        else if (upload.isFlagged != 1) {

```

Comment:
If there is no way around having 3-4 if statements then we should introduce switch statements instead